web/src/Legal.tsx

```diff
@@ -0,0 +1,181 @@
+// This file is part of tacd, the LXA TAC system daemon
+// Copyright (C) 2024 Pengutronix e.K.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import Container from "@cloudscape-design/components/container";
+import Header from "@cloudscape-design/components/header";
+import Table from "@cloudscape-design/components/table";
+import Link from "@cloudscape-design/components/link";
+import SpaceBetween from "@cloudscape-design/components/space-between";
+
+import { useEffect, useState } from "react";
+
+type Package = {
+  package_name: string;
+  version: string;
+  recipe_name: string;
+  license: string;
+};
+
+function parse_mainifest(text: string) {
+  let packages: Package[] = [];
+
+  // The content of `text` looks something like this:
+  //
+  // PACKAGE NAME: tacd
+  // PACKAGE VERSION: 1.0.0
+  // RECIPE NAME: tacd
+  // LICENSE: GPL-2.0-or-later
```

Comment:
How stable is the format generated by [`poky/meta/classes-recipe/license_image.bbclass`](https://github.com/yoctoproject/poky/blob/master/meta/classes-recipe/license_image.bbclass#L85-L88)? How hard would it be to have some kind of unit-test against a recent license file?